src/vip/batch_address_test_tool/processor.clj

```diff
@@ -63,7 +63,10 @@
 
 (defn retrieve-polling-locations*
   [ctx]
-  ctx)
+  (let [addresses (:addresses ctx)
+        polling-locations (doall (map #(civic-info/address->polling-location (:address %)) addresses))
+        merged (mapv #(assoc %1 :api-result %2) addresses polling-locations)]
```

Comment:
We're not using `mapv` to produce a vector, `mapv` is applying the function to the first element of both collections, in this case, it's getting the address hashmap from the first collection and the corresponding polling-location from the second collection and `assoc`ing it into the address hashmap, essentially building adding the result into it. Maybe it'd be clearer with a different name than `addresses`? Each one represents a mini-context for a row of data.